### PR TITLE
[stdlib] create os.remove and os.unlink

### PR DESCRIPTION
**Pull request by artemiogr97**
Implementation of #2306

### CODE REVIEW

File: stdlib/src/os/os.mojo
Comment (JoeLoser) [inline stdlib/src/os/os.mojo:253]: **Quibble** `s/speficied/specified`

Comment (JoeLoser) [inline stdlib/src/os/os.mojo:253]: **Suggestion** Be clear about the semantics about what happens if the argument passed is a path to a directory.  In Python, according to their [docs](https://docs.python.org/3/library/os.html#os.remove), if a directory is passed, an error is raised.

Pedantically, should we also spell out whether the path is relative or absolute?

File: stdlib/src/os/os.mojo
Comment (JoeLoser) [inline stdlib/src/os/os.mojo:259]: **Suggestion** It doesn't matter too much in practice, but I'd use `Int` instead of `Int8` here as it more closely matches the UNIX function return type as seen [here](https://man7.org/linux/man-pages/man2/unlink.2.html).  This would follow the element of least surprise to future readers even though the return values all technically fit in the type domain of an `Int8` as you have it now.

File: stdlib/src/os/os.mojo
Comment (JoeLoser) [inline stdlib/src/os/os.mojo:280]: Ditto here about the small typo as above.

File: stdlib/test/os/test_remove.mojo
Comment (JoeLoser) [inline stdlib/test/os/test_remove.mojo:13]: **Suggestion** Change this to just `RUN: %mojo-no-debug %s` as we recently dropped the `-debug-level full` stuff in our stdlib tests in https://github.com/modularml/mojo/commit/b6a8f871174bde8211b43bfec50a5a7104e116df

File: stdlib/src/os/os.mojo
Comment (JoeLoser) [inline stdlib/src/os/os.mojo:262]: **Suggestion** Do you mind filing a GitHub issue to support capturing the error message on failure and link to it here in the code?  This will help others who are interested in improving this to be able to cross-reference things.

**Review by JoeLoser — COMMENTED** [review]
Generally looks good, thank you for the contribution!  Just a few minor comments about being more specific about what's currently supported, etc.  Once that is addressed, I'm happy to merge this!

**Comment by artemiogr97** [comment]
Sure, I'll check that later today

File: stdlib/src/os/os.mojo
Comment (artemiogr97) [reply stdlib/src/os/os.mojo:267]: I'll wait for the merge to create the issue, to be able to reference the code in the nightly branch

**Review by JoeLoser — APPROVED** [review]

File: stdlib/src/os/os.mojo
Comment (abduld) [reply stdlib/src/os/os.mojo]: This should be Int32 since it matches the C int type

**Pull request merged**
